### DISCUSSION TITLE
GTFS-RT using gtfs catalog item

### DISCUSSION THREAD

**Discussion by imakihi** [TerriaJS/terriajs#5543] [Q&A] [answered]
Hi there.
I am trying to visualize real time bus and train data around the Tokyo metro area like this NSW example.
https://nsw.digitaltwin.terria.io/#share=s-wOjVtLuQ0cDs55DLx5ElKhBZyiV

 I wanted to use GTFS Catalog Item (https://docs-v8.terria.io/guide/connecting-to-data/catalog-type-details/gtfs/) but I couldn't make it work. The data source is GTFS-RT (https://developers.google.com/transit/gtfs-realtime/). I could read data with my python code, so the server and data themselves are good. 

Can Terria read GTFS-RT directory using the GTFS catalog item?

This is my error message.

Could not load Bus GTFS.
There was an error loading the data for route　GTFS.

GTFS data source info is here
https://ckan.odpt.org/dataset/b_bus_gtfs_rt-toei/resource/a5bb2966-d4f3-474e-aaf0-941a3ca6a662

My catalog is like this
`		{
			"name":"Bus GTFS",
			"type":"gtfs",
			"url":"https://api.odpt.org/api/v4/gtfs/realtime/ToeiBus?acl:consumerKey=xxxx",
			"refreshInterval": 5
		},`

If Terria can consume GTFS-RT data, then I am doing something wrong.

Can anyone help me to understand how to visualize GTFS-RT data?

Thank you in advance.

Hiroo

**Comment by hardreddata** [comment, marked as answer]
Hi @imakihi 

You need to be using the `next` branch of Terria for this to work.

Take a look at this example https://github.com/TerriaJS/TerriaMap/blob/next/wwwroot/init/simple.json#L162-L180 which you get when you clone `next`.

Cheers.

**Reply by imakihi** [reply]
Hi @RussellGrew. Thank you so much for your response. I updated my Terria to the next branch and I have been trying to show data. So far, I couldn't show my data source yet but I am going to try a bit more.

I realized that after I installed the next branch, I couldn't show the NSW Live Transport - Buses example neither. So, I may be encountering a different problem instead of data connection...

I am getting a following error message... 

`Resource.js?666e:2366 GET https://api.odpt.orghttps//api.transport.nsw.gov.au/v1/gtfs/vehiclepos/buses net::ERR_NAME_NOT_RESOLVED`

Thank you.
Hiroo

**Reply by hardreddata** [reply]
Looking at `https://api.odpt.orghttps//api.transport.nsw.gov.au/v1/gtfs/vehiclepos/buses` you have somehow got two URLs in there?

You can see the buses working here http://ci.terria.io/next/

**Reply by imakihi** [reply]
@RussellGrew 
Thank you for pointing that out.  

I basically just copied the NSW bus catalog item to my catalog and got the error. I just copied the error message from my developer too, so I may not copy the error message correctly.

So far, I converted data to CSV and am refreshing my csv file every 5sec. This works okay but not sophisticated since Terria include GTFS catalog item. I will spend more time to figure this out.

Thank you for helping!

Hiroo
